plugins/nanospell/plugin.js

```diff
@@ -590,24 +594,37 @@
 			 for a given range, get the unique words in it that we don't have a spellcheck status for
 			 */
 			function scanWordsInRange(range) {
-				var block;
-				var iterator = range.createIterator();
+				var combinedText = '',
+					block,
+					blockList = [],
+					iterator = range.createIterator();
 				while (( block = iterator.getNextParagraph() )) {
 					block.setCustomData('spellCheckInProgress', true);
-					var unknownWords = getUnknownWords(block.getText());
-					startCheckOrMarkWords(unknownWords, block);
+					combinedText += block.getText() + ' ';
```

Comment:
Interesting.  One thing I want to test my understanding on:

If we have nested paragraph-like items, the words will be duped here in combinedText, however, the list of words will be made unique later on.